syntax = "proto3";

service MLTaskService {
  rpc ExecuteTask(TaskRequest) returns (TaskResponse);
}

message TaskRequest {
  string algorithm = 1;
  string dataset_id = 2;
  map<string, string> hyperparameters = 3;
}

message TaskResponse {
  string worker_id = 1;
  string result = 2;
  double execution_time = 3;
}
